### DISCUSSION TITLE
Add LaTeX package from a Lua filter

### DISCUSSION THREAD

**Discussion by bgrenet** [jgm/pandoc#10361] [Q&A] [answered]
I'd like to know whether it is possible to add a LaTeX package using a Lua filter. 

One example of use case: I have a Lua filter that renders some tables using the environment `tblr` of the LaTeX package tabularray. To make the conversion from markdown to PDF, one needs to put a `usepackage{tabularray}` in the preamble. Of course, this can be set in the document metadata, or using a custom template. But I'd like that simply saying `pandoc document.md --lua-filter=tblr.lua -o document.pdf` is enough. Is it possible?

(For this specific case, there may be a workaround. Although I am interested in learning such a workaround, I'm still interested in my original question since I regularly hit this question.)

**Comment by jgm** [comment, marked as answer]
You can add the `usepackage` command to a `header-includes` field in metadata.
Caveat:  this won't work well if you're also using `--include-in-header` on the command line (as this will overwrite anything in metadata).

**Comment by bgrenet** [comment]
Indeed, I guess that is the way for me. I can live without `--include-in-header` anyway.
